src/meta/src/manager/catalog/mod.rs

```diff
@@ -271,44 +271,61 @@ where
         }
     }
 
-    pub async fn start_create_procedure(&self, relation: &Relation) -> MetaResult<()> {
-        match relation {
-            Relation::Table(table) => self.start_create_table_procedure(table).await,
-            Relation::Sink(sink) => self.start_create_sink_procedure(sink).await,
-            Relation::Index(index, index_table) => {
+    pub async fn start_create_stream_job_procedure(
+        &self,
+        stream_job: &StreamingJob,
+    ) -> MetaResult<()> {
+        match stream_job {
+            StreamingJob::Table(table) => self.start_create_table_procedure(table).await,
+            StreamingJob::Sink(sink) => self.start_create_sink_procedure(sink).await,
+            StreamingJob::Index(index, index_table) => {
                 self.start_create_index_procedure(index, index_table).await
             }
+            StreamingJob::MaterializedSource(source, table) => {
+                self.start_create_materialized_source_procedure(source, table)
+                    .await
+            }
         }
     }
 
-    pub async fn cancel_create_procedure(&self, relation: &Relation) -> MetaResult<()> {
-        match relation {
-            Relation::Table(table) => self.cancel_create_table_procedure(table).await,
-            Relation::Sink(sink) => self.cancel_create_sink_procedure(sink).await,
-            Relation::Index(index, index_table) => {
-                self.cancel_create_index_procedure(index, index_table).await
-            }
+    pub async fn mark_creating_tables(&self, creating_tables: &[Table]) {
+        let core = &mut self.core.lock().await.database;
+        core.mark_creating_tables(creating_tables);
+        for table in creating_tables {
+            self.notify_compute_and_compactor(Operation::Add, Info::Table(table.to_owned()))
+                .await;
         }
```

Comment:
IIUC, this is for prefix filter, and the frontends should not know about this creating materialized views until the process finishes?

Reply:
yes, the creating status of materialized views (and internal table) only be notified to compute and compactor (for prefix filter)